wp/lib/bap_wp/src/z3_utils.ml

```diff
@@ -111,3 +111,58 @@ let mk_smtlib2_single (env : Env.t) (smt_post : string) : Constr.t =
   let ctx = Env.get_context env in
   mk_smtlib2 ctx smt_post decl_syms
 
+let construct_pointer_constraint (l: string list) (env1 : Env.t)
+    (env2: Env.t option): Constr.t =
+  let stack_bottom = Env.get_stack_bottom env1 in
+  let ctx = Env.get_context env1 in
+  let arch = match Env.get_arch env1 |> Bap.Std.Arch.addr_size with
+    | `r32 -> 32
+    | `r64 -> 64 in
+  let rsp = "RSP" in
```

Comment:
Hardcoding the string "RSP" makes me a bit uncomfortable. Since we have the architecture, we can grab the name of the stack pointer:
```
let module Target = (val target_of_arch arch) in
let rsp = Var.name Target.CPU.sp in
...
```